### DISCUSSION TITLE
House Total Energy (Solax X3 G4)

### DISCUSSION THREAD

**Discussion by ZdenekM** [wills106/homeassistant-solax-modbus#606] [Q&A] [answered]
Hi all. Is there a way how to get "house total energy" with Solax? I mean solar yield + grid import. The total that the household consumed regardless of the source, like an integral of `sensor.solax_house_load`. It seems from the code there is `house_load_total` for some inverters, but not for Solax.

**Comment by cleveHEX** [comment, marked as answer]
You can calculate the entity from provided entities.

**Reply by ZdenekM** [reply]
Yeah, I would, but... There should probably be `grid_import_total`, but it's not in my case :-/ What could be the reason? Otherwise it would be `grid_import_total+total_yield`, right?

**Reply by cleveHEX** [reply]
Other inverters could offer the information, SolaX does not, at least I did not find any address for it. I did not study the behaviour of all entities deeply, but you could be missing the energy you sell back to grid, but I am not sure about that (probably needs to be subtracted).

**Reply by ZdenekM** [reply]
`grid_import_total` should be there (probably) for Solax - I found it here https://github.com/wills106/homeassistant-solax-modbus/blob/e6dd47b38976ea37eb00308eb9f20ed92d8da2f1/custom_components/solax_modbus/plugin_solax.py#L3686.

**Reply by ZdenekM** [reply]
Aha, there is `entity_registry_enabled_default=False` for this one... Is there a way to enable it?

**Reply by ZdenekM** [reply]
Ok, that was pretty easy... Can be done through UI.

**Reply by ZdenekM** [reply]
So I will go on with `{{ states("sensor.solax_total_yield") | float + states("sensor.solax_grid_import_total") | float - states("sensor.solax_grid_export_total") | float }}`. Hopefully it is correct :) Thanks.

**Reply by cleveHEX** [reply]
Do you have HA energy dashboard? You can compare the data with this.
